TrigScint/include/TrigScint/TrigScintFirmwareHitProducer.h

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file TrigScintFirmwareHitProducer.h
+ * @brief Staging of Real Hits
+ * @author Lene Kristian Bryngemark, Stanford University
+ */
+
+#ifndef TRIGSCINT_TRIGSCINTFIRMWAREHITPRODUCER_H
+#define TRIGSCINT_TRIGSCINTFIRMWAREHITPRODUCER_H
+
+/*~~~~~~~~~~*/
+/*   ROOT   */
+/*~~~~~~~~~~*/
+#include "TRandom3.h"
+
+// LDMX
+#include "DetDescr/TrigScintID.h"
+#include "Recon/Event/EventConstants.h"
+#include "Tools/NoiseGenerator.h"
+#include "TrigScint/Event/TrigScintHit.h"
+#include "TrigScint/Event/TrigScintQIEDigis.h"
+
+/*~~~~~~~~~~~~~~~*/
+/*   Framework   */
+/*~~~~~~~~~~~~~~~*/
+#include "Framework/Configure/Parameters.h"
+#include "Framework/EventProcessor.h"
+
+/*~~~~~~~~~~~*/
+/* TrigScint */
+/*~~~~~~~~~~~*/
+#include "TrigScint/Firmware/objdef.h"
+#include "TrigScint/SimQIE.h"
+
+namespace trigscint {
+
+/**
+ * @class TrigScintFirmwareHitProducer
+ * @brief
+ */
+class TrigScintFirmwareHitProducer : public framework::Producer {
+ public:
+  TrigScintFirmwareHitProducer(const std::string& name,
+                               framework::Process& process)
+      : Producer(name, process) {}
+
+  void configure(framework::config::Parameters& ps) override;
+
+  void produce(framework::Event& event) override;
+
+  /**
+   * add a hit at index idx to a cluster
+   */
+
+ private:
+  /// Class to set the verbosity level.
+  // TODO: Make use of the global verbose parameter.
+  bool verbose_{false};
```

Comment:
Like other comment said - goal is to remove this as a member variable and put any high-verbosity messages into the `ldmx_log(debug)` channel.

Reply:
done :)!